### DISCUSSION TITLE
API requests fail to use HTTP proxy

### DISCUSSION THREAD

**Discussion by mindspawn** [open-webui/open-webui#2800] [General]
# Bug Report

## Description

**Bug Summary:**
HTTP requests are not using the environment http_proxy

I believe I have resolved the issue and will raise a PR soon.

**Steps to Reproduce:**
Set http_proxy and requests do not go to the proxy

**Expected Behavior:**
Proxy is used

**Actual Behavior:**
Proxy is not used

**Comment by justinh-rahb** [comment]
We don't have an `http_proxy` environment variable, also WebUI v0.1.120 is very old by now, we only support latest `main` version anyway.

**Reply by mindspawn** [reply]
It's not a open-webui environment variable. It is a variable used both on linux and windows to specify a proxy.
The Python requests library honors it by default. However, the ClientSession method from aiohttp does not honor it by default.
I have fixed this issue in: https://github.com/open-webui/open-webui/pull/2799

**Comment by mlier** [comment]
I think you missed to change a ClientSession in open-webui/backend/main.py ligne 900 to check if there is an updated version.

```
@app.get("/api/version/updates")
async def get_app_latest_release_version():
    try:
        async with aiohttp.ClientSession(trust_env=True) as session:
            async with session.get(
                "https://api.github.com/repos/open-webui/open-webui/releases/latest"
            )
```

**Reply by mindspawn** [reply]
Thanks. Good catch your right. Will fix.

**Reply by mindspawn** [reply]
Raised https://github.com/open-webui/open-webui/pull/2870
